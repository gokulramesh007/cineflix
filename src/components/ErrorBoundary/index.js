import LotteryErrorBoundary from "./LotteryErrorBoundary";

export { LotteryErrorBoundary };
