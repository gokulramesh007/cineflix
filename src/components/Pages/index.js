import Booking from "./Booking/Booking";
import MovieDescription from "./MovieDescription/MovieDescription";
import MovieList from "./MovieList/MovieList";

export { Booking, MovieDescription, MovieList };
