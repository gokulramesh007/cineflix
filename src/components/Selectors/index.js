import FoodSelector from "./FoodSelector/FoodSelector";
import LanguageSelector from "./LanguageSelector/LanguageSelector";
import ParkingSelector from "./ParkingSelector/ParkingSelector";

export { FoodSelector, LanguageSelector, ParkingSelector };
