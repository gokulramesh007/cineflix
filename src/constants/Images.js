const images = {
  BIKE: require("../assets/bike.svg"),
  BURGER: require("../assets/burger.svg"),
  CAR: require("../assets/car.svg"),
  PIZZA: require("../assets/pizza.svg"),
  TANDOORI: require("../assets/turkey.svg"),
  ERROR: require("../assets/404-error.jpg"),
  LOADER: require("../assets/loader.gif"),
  POSTER_NOT_FOUND: require("../assets/poster-unavailable.png"),
  POSTER_COMING_SOON: require("../assets/movie-poster-coming-soon.png"),
  IMAGE_UNAVAILABLE: require("../assets/no-image-available.png")
};

export default images;
