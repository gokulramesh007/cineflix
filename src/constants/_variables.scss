$mobile-width: 990px;

// @mixin mobile {
//   @media only screen and (max-width: #{$mobile-width}) {
//     @content;
//   }
// }

/* Color Variables Declaration - START */

$white: #fff;
$black: #444;
$container-green: #3ade26;
$button-purple: rgb(141,21,251);
$button-yellow: #f3a619;
$loader-overlay: rgba(246, 246, 246, 0.2);

/* Color Variables Declaration - END */

/* Font Variables Declaration - START */

$extraLarge: 70px;
$h1: 50px;
$h2: 40px;
$h3: 30px;
$h4: 26px;
$large: 23px;
$h5: 20px;
$h6: 19px;
$input: 18px;
$regular17: 17px;
$regular: 16px;
$regular15: 15px;
$medium: 14px;
$smallMedium: 13px;
$small: 12px;
$smaller: 10px;
$tiny: 8px;
$minute: 6px;
$space_large: 7px;

/* Font Variables Declaration - END */
