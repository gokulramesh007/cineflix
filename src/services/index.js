import lotteryService from "./lotteryService";
import movieService from "./movieService";
import reserveService from "./reserveService";

export { lotteryService, movieService, reserveService };
