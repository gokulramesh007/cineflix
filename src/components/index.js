import Button from "./Button/Button";
import { MovieCard } from "./Card";
import Container from "./Container/Container";
import { LotteryErrorBoundary } from "./ErrorBoundary";
import Image from "./Image/Image";
import Loader from "./Loader/Loader";
import { MoviesList } from "./List";
import { Booking, MovieDescription, MovieList } from "./Pages";
import Promotions from "./Promotions/Promotions";
import { FoodSelector, LanguageSelector, ParkingSelector } from "./Selectors";
import Tile from "./Tile/Tile";

export {
  Booking,
  Button,
  Container,
  FoodSelector,
  Image,
  LanguageSelector,
  Loader,
  LotteryErrorBoundary,
  MovieCard,
  MovieDescription,
  MovieList,
  MoviesList,
  ParkingSelector,
  Promotions,
  Tile
};
