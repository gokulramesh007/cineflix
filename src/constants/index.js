import Images from "./Images";
import Messages from "./Messages";
import Strings from "./Strings";
import moviesMock from "./Mock/MovieServiceMock";
import { preferenceMock } from "./Mock/ReserveServiceMock";

export { Images, Messages, moviesMock, preferenceMock, Strings };
