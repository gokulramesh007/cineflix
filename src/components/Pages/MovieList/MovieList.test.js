import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";
import { movieService } from "../../../services";
import { moviesMock } from "../../../constants";
import { MovieList } from "../../../components";

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("render with no data", async () => {
  await act(async () => {
    render(<MovieList />, container);
  });

  expect(pretty(container.innerHTML)).toMatchSnapshot();
});

it("render with mock data", async () => {
  const movies = await movieService.getMovies();
  //console.log(movies);
  expect(movies[0].title).toEqual("Anand");

  jest
    .spyOn(movieService, "getMovies")
    .mockImplementation(() => Promise.resolve(moviesMock.data));

  await act(async () => {
    render(<MovieList />, container);
  });

  expect(pretty(container.innerHTML)).toMatchSnapshot();

  movieService.getMovies.mockRestore();
});
