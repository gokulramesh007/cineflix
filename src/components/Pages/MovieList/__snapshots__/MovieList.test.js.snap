// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with mock data 1`] = `
"<div class=\\"tileWrapper \\">
  <div class=\\"title\\">All Movies</div>
  <div class=\\"movieListWrapper\\">
    <div class=\\"product-list-wrapper two\\">
      <div class=\\"tiles\\">
        <div class=\\"movieCardWrapper\\">
          <div class=\\"imageWrapper\\"><img class=\\"image\\" src=\\"https://images-na.ssl-images-amazon.com/images/M/MV5BMjE0Mzk3OTk2NF5BMl5BanBnXkFtZTgwMTQ1NDk5NTE@._V1_SY250_CR0,0,187,250_AL_.jpg\\" alt=\\"Anand\\"></div>
          <div class=\\"details\\">
            <div class=\\"title\\">Anand</div>
            <div class=\\"info\\">
              <div class=\\"visit\\">0 Visits</div>
              <div class=\\"like\\">0 Likes</div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"hide\\">
    <div class=\\"loader-wrapper\\">
      <div class=\\"loader\\"><img src=\\"loader.gif\\" alt=\\"Loading...\\"></div>
    </div>
  </div>
  <div class=\\"buttonWrapper\\"><button class=\\"button yellow dark custom\\">LOAD MORE</button></div>
</div>"
`;

exports[`render with no data 1`] = `
"<div class=\\"tileWrapper \\">
  <div class=\\"title\\">All Movies</div>
  <div class=\\"movieListWrapper\\">
    <div class=\\"product-list-wrapper two\\"></div>
  </div>
  <div>
    <div class=\\"loader-wrapper\\">
      <div class=\\"loader\\"><img src=\\"loader.gif\\" alt=\\"Loading...\\"></div>
    </div>
  </div>
  <div class=\\"buttonWrapper\\"><button class=\\"button yellow dark custom\\">LOAD MORE</button></div>
</div>"
`;
